packages/x-date-pickers/src/DesktopDatePicker/DesktopDatePickerKeyboard.test.tsx

```diff
@@ -46,6 +46,24 @@ describe('<DesktopDatePicker /> keyboard interactions', () => {
     expect(handleClose.callCount).to.equal(1);
   });
 
+  it('does not trigger onClose on Escape press if already closed', () => {
```

Comment:
Could you please put it in the `picker lifecycle` `describe` and follow the structure of the test there ("should not call onClose or onAccept when clicking outside of the picker if not opened" being a good candidate testing the exact same behavior).

That way we also test that `onAccept` and `onChange` are not called.